### PR TITLE
Support quit command

### PR DESCRIPTION
**Pull request by mckelvin**
- quit: Call `quit()` to close all TCP connections of a client.

@zzl0 Please review.

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -150,6 +150,13 @@ err_code_t Client::version(types::broadcast_result_t** results, size_t* nHosts)
 }
 
 
+err_code_t Client::quit() {
+  broadcastCommand(keywords::kQUIT, 4);
```
Comment (zzl0) [inline src/Client.cpp]: Why we pass the number `4`(since it can be computed by `strlen(keywords::kQUIT)`)  explicitly, is this an optimization for performance ?

Comment (mckelvin) [inline src/Client.cpp]: The inner interface([`BufferWriter::takeBuffer(const char* const buf, size_t buf_len)`](https://github.com/douban/libmc/blob/9ed101bcde3556adbb5a1228bcf2805ddd8e8129/src/BufferWriter.cpp#L38)) is designed to be compatible with arbitrary string buffer.  (e.g.: if `\0` is in a buffer, we can never call `strlen` to get the actual length.)

Comment (zzl0) [inline src/Client.cpp]: I mean may be we can use `strlen` in function `broadcastCommand`.

**Comment by zzl0** [comment]
other :lgtm:

File: src/Client.cpp (hunk 1 above)
Comment (mckelvin) [inline src/Client.cpp]: I would prefer not to call `strlen` for a **static const variable** at runtime. May be it can be optimised at compile time but I'm not sure.

Comment (zzl0) [inline src/Client.cpp]: ok.  http://stackoverflow.com/a/21067617/811777

Comment (mckelvin) [inline src/Client.cpp]: `-O3` is included at compile time by default, I followed the advice. Thanks.

Comment (lembacon) [inline src/Client.cpp]: Technically, relying on the compiler optimizations is not always a good practice.

You should take advantages of C++ templates to get the length of string literals during compile time.

```cpp
template <typename _CharType, size_t _Length>
inline size_t get_string_length(const _CharType (&)[_Length])
{
  return _Length - 1;
}
```

Comment (mckelvin) [inline src/Client.cpp]: @lembacon @zzl0  This changeset is excluded now since it's OT, not so emergent, and there're more than one solution. Would please send the patch in another PR ?

Comment (zzl0) [inline src/Client.cpp]: :lgtm:
